examples/e-commerce/src/app/app.component.ts

```diff
@@ -11,4 +11,24 @@ export class AppComponent {
     indexName: 'instant_search',
     routing: true,
   };
+
+  onKeyUp = event => {
+    if (event.key !== 'Escape') {
+      return;
+    }
+    this.closeFilters();
+  };
+
+  public openFilters() {
+    document.body.classList.add('filtering');
+    window.scrollTo(0, 0);
+    window.addEventListener('keyup', this.onKeyUp);
+  }
+
+  public closeFilters() {
+    document.body.classList.remove('filtering');
+    const resultsContainer = document.querySelector('.container-results');
+    resultsContainer.scrollIntoView();
```

Comment:
Comment about all InstantSearch flavours mobile demos:
Would it make sense to also scroll up when the pagination button is clicked?

Right now, it feels weird that you have to browse through the hits in the inverted order after switching to the next page. 

![2019-06-26 11 57 17](https://user-images.githubusercontent.com/2982512/60170841-98f79200-9809-11e9-8372-68472facec63.gif)

Reply:
This is the behavior in InstantSearch.js. Vue and Angular don't provide a built-in way to do so but we should try to integrate it if it doesn't create to many complications.

In React InstantSearch, we used to use the [`ScrollTo`](https://www.algolia.com/doc/api-reference/widgets/scroll-to/react/) component but it didn't play well with filtering on mobile (it scrolled to the container that was hidden).

Definitely an improvement that would be nice to have!